script/SingletonDeployer_v1.s.sol

```diff
@@ -0,0 +1,275 @@
+
+    //--------------------------------------------------------------------------
+    // Factory Usage
+
+    // Deploy the Governor_v1.
+    /*
+    // Deploy without Arguments
+    bytes memory implBytecode = vm.getCode("Governor_v1.sol:Governor_v1");
+    govImplementation = factory.deployWithCreate2(salt, implBytecode);
+
+    // Deploy with Arguments
+    bytes memory proxyArgs = abi.encode(address(govImplementation), communityMultisig, bytes(""));
+    bytes memory proxyCode = vm.getCode("TransparentUpgradeableProxy.sol:TransparentUpgradeableProxy");
+    bytes memory proxyBytecode = abi.encodePacked(proxyCode, proxyArgs);
+    gov = Governor_v1(
+        address(
+                factory.deployWithCreate2(salt, proxyBytecode)
+        )
+    ); */
+
+    function createExternalSingletons() public {
+        console2.log(
+            "--------------------------------------------------------------------------------"
+        );
+        console2.log("Create External Singletons");
+        console2.log("-External Contracts");
+
+        //External contracts
+        ext_InverterReverter_v1 = factory.deployWithCreate2( //@note no Beacon structure here right?
+        factorySalt, vm.getCode("InverterReverte_v1.sol:InverterReverter_v1")); //@note can we add a string parameter for the factory? So we can log which contract is called here?
+
+        ext_Governor_v1 = factory.deployWithCreate2(
+            factorySalt, vm.getCode("Governor_v1.sol:Governor_v1")
+        );
+
+        ext_TransactionForwarder_v1 = factory.deployWithCreate2(
+            factorySalt,
+            vm.getCode("TransactionForwarder_v1.sol:TransactionForwarder_v1")
+        );
+
+        ext_FeeManager_v1 = factory.deployWithCreate2(
+            factorySalt, vm.getCode("FeeManager_v1.sol:FeeManager_v1")
+        );
+    }
+
+    function createWorkflowAndFactorySingletons(address transactionForwarder)
+        public
+    {
+        console2.log(
+            "--------------------------------------------------------------------------------"
+        );
+        console2.log("Create Workflow and Factory Singletons");
+
+        //Factories
+        console2.log("-Factories");
+
+        fac_ModuleFactory_v1 = factory.deployWithCreate2(
+            factorySalt,
+            abi.encodePacked(
+                vm.getCode("ModuleFactory_v1.sol:ModuleFactory_v1"),
+                abi.encode(ext_InverterReverter_v1, transactionForwarder)
+            )
+        );
+
+        fac_OrchestratorFactory_v1 = factory.deployWithCreate2(
+            factorySalt,
+            abi.encodePacked(
+                vm.getCode("OrchestratorFactory_v1.sol:OrchestratorFactory_v1"),
+                abi.encode(ext_InverterReverter_v1, transactionForwarder)
+            )
+        );
+
+        //Modules
+        console2.log("-Modules");
+
+        //Authorizer
+        console2.log("--Authorizer");
+
+        mod_Aut_Roles_v1 = factory.deployWithCreate2(
```

Comment:
for readability, I'd mention in the variable names that these are the implementations. What do you think?